app/models/models.py

```diff
@@ -39,102 +43,137 @@ def hashed_password(password):
-                    on t.user_id = u.email""") # join with users table
-                    
-        for t in result:
-            if t.user_id in user_to_task:
-                user_to_task.get(t.user_id).append(dict(t))
-            else:
-                user_to_task[t.user_id] = [dict(t)]
+        users = user_container.query_items(
+            query="SELECT * FROM users", enable_cross_partition_query=True
+        )
+
+        cnt = 0
+        for u in users:
+            cnt += 1
+            tasks = task_container.query_items(
+                query="SELECT * FROM tasks t WHERE t.user_id = @userId",
+                parameters=[dict(name="@userId", value=u["id"])],
+                enable_cross_partition_query=True,
+            )
+            for t in tasks:
+                t["first_name"] = u["first_name"]
+                t["last_name"] = u["last_name"]
+                if t["user_id"] in user_to_task:
+                    user_to_task.get(t["user_id"]).append(t)
+                else:
+                    user_to_task[t["user_id"]] = [t]
 
         return user_to_task
 
     @staticmethod
     def get_tasks_for_user(user_id):
-        return Task.query.filter_by(user_id=user_id)
+        tasks = task_container.query_items(
+            query="SELECT * FROM tasks t WHERE t.user_id = @userId",
+            parameters=[dict(name="@userId", value=user_id)],
+            enable_cross_partition_query=True,
+        )
+        return tasks
 
     @staticmethod
     def delete_task(task_id):
-        task_to_delete = Task.query.filter_by(id=task_id).first()
-        db.session.delete(task_to_delete)
-
         try:
-            db.session.commit()
+            tasks = task_container.query_items(
+                query="SELECT * FROM tasks t WHERE t.id = @taskId",
+                parameters=[dict(name="@taskId", value=task_id)],
+                enable_cross_partition_query=True,
+            )
+            task = list(tasks)[0]
+            task_container.delete_item(task, partition_key=task_id)
+
             return True
-        except IntegrityError:
+        except exceptions.CosmosHttpResponseError:  # item wasn't deleted successfully
+            return False
+        except exceptions.CosmosResourceNotFoundError:  # item wasn't found
             return False
-    
+
     @staticmethod
     def edit_task(task_id, task, status):
-        task_to_edit = Task.query.filter_by(id=task_id).first()
-        task_to_edit.task = task
-        task_to_edit.status = status
-
         try:
-            db.session.commit()
+            task_to_edit = task_container.read_item(task_id, partition_key=task_id)
+            task_to_edit["task"] = task
+            task_to_edit["status"] = status
+
+            task_container.upsert_item(task_to_edit)
+
             return True
-        except IntegrityError:
+        except exceptions.CosmosHttpResponseError:  # item couldn't be upserted
             return False
```

Comment:
Let's do some research on how we want to handle error messages from the model. Typically the UI doesn't want to show the error details, but something to consider, versus just returning false.

Reply:
Oh the UI wouldn't just show 'False.' It'll have an error message that's appropriate for the specific situation. Please take a look at the routes.py script.